xdsl/dialects/builtin.py

```diff
@@ -627,48 +635,93 @@ def shape_is_complete(self) -> bool:
 
     @staticmethod
     def create_dense_index(
-            type: VectorOrTensorOf[IndexType],
-            data: List[int | IntegerAttr[IndexType]]
+        type: VectorOrTensorOf[IndexType],
+        data: Sequence[int] | Sequence[IntegerAttr[IndexType]]
     ) -> DenseIntOrFPElementsAttr:
-        attr_list = [
-            IntegerAttr.from_index_int_value(d) if isinstance(d, int) else d
-            for d in data
-        ]
+        if len(data) and isinstance(data[0], int):
+            attr_list = [
+                IntegerAttr.from_index_int_value(d)
+                for d in cast(Sequence[int], data)
+            ]
+        else:
+            attr_list = cast(Sequence[IntegerAttr[IndexType]], data)
+
         return DenseIntOrFPElementsAttr([type, ArrayAttr(attr_list)])
 
     @staticmethod
     def create_dense_int(
         type: VectorOrTensorOf[IntegerType],
-        data: List[int | IntegerAttr[IntegerType]]
+        data: Sequence[int] | Sequence[IntegerAttr[IntegerType]]
     ) -> DenseIntOrFPElementsAttr:
-        attr_list = [
-            IntegerAttr.from_params(d, type.element_type) if isinstance(
-                d, int) else d for d in data
-        ]
+        if len(data) and isinstance(data[0], int):
+            attr_list = [
+                IntegerAttr[IntegerType](d, type.element_type)
+                for d in cast(Sequence[int], data)
+            ]
+        else:
+            attr_list = cast(Sequence[IntegerAttr[IntegerType]], data)
+
         return DenseIntOrFPElementsAttr([type, ArrayAttr(attr_list)])
 
     @staticmethod
     def create_dense_float(
-            type: VectorOrTensorOf[AnyFloat],
-            data: List[int | float | AnyFloatAttr]
+        type: VectorOrTensorOf[AnyFloat],
+        data: Sequence[int | float] | Sequence[AnyFloatAttr]
     ) -> DenseIntOrFPElementsAttr:
-        data_attr = [
-            FloatAttr(float(d), type.element_type)
-            if not isinstance(d, FloatAttr) else d for d in data
-        ]
-        return DenseIntOrFPElementsAttr([type, ArrayAttr(data_attr)])
+        if len(data) and isinstance(data[0], int | float):
+            attr_list = [
+                FloatAttr(float(d), type.element_type)
+                for d in cast(Sequence[int | float], data)
+            ]
+        else:
+            attr_list = cast(Sequence[AnyFloatAttr], data)
 
+        return DenseIntOrFPElementsAttr([type, ArrayAttr(attr_list)])
+
+    @overload
```

Comment:
Can we fuse this one and the next one? I'm not sure whey they are separate.

Reply:
You're right
- [ ] fuse with overload below